pkg/connection/connection.go

```diff
@@ -650,6 +653,26 @@ func (m *Manager) AddBqConnectionFromConfig(connection *config.GoogleCloudPlatfo
 	}
 	m.mutex.Unlock()
 
+	if len(connection.ServiceAccountFile) > 0 && connection.ServiceAccountFile != "" {
+		file, err := ioutil.ReadFile(connection.ServiceAccountFile)
+		if err != nil {
+			return errors.New("failed to read service account file")
+		}
+		var js json.RawMessage
+		err = json.Unmarshal(file, &js)
+		if err != nil {
+			return errors.New("not a valid JSON in service account file")
+		}
+	}
+
+	if len(connection.ServiceAccountJSON) > 0 && connection.ServiceAccountJSON != "" {
+		var js json.RawMessage
+		err := json.Unmarshal([]byte(connection.ServiceAccountJSON), &js)
+		if err != nil {
+			return errors.New("not a valid JSON in service account json")
```

Comment:
remember to include here a check according to the ticket if it looks like a file path, it can recommend sth like: would you like to use service_account_file instead?

Reply:
I just updated all error messages and please let me know

Reply:
the idea is these two fields are confusing:
```
service_account_json:
service_account_file:
```

therefore sometimes users confuse these two, and give a file path to the JSON. we could maybe check if that looks like a path and return a custom error?